server/src/main/java/org/elasticsearch/action/admin/cluster/state/ClusterStateResponse.java

```diff
@@ -103,5 +118,26 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_6_0_0_alpha1)) {
             totalCompressedSize.writeTo(out);
         }
+        // TODO: change version to V_6_6_0 after backporting:
+        if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
+            out.writeBoolean(timedOut);
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ClusterStateResponse response = (ClusterStateResponse) o;
+        return timedOut == response.timedOut &&
+            Objects.equals(clusterName, response.clusterName) &&
+            // Best effort. Left out cluster state, because it doesn't implement equals() and hashcode()
+            Objects.equals(totalCompressedSize, response.totalCompressedSize);
+    }
+
+    @Override
+    public int hashCode() {
+        // Best effort for testing. Left out cluster state, because it doesn't implement equals() and hashcode()
```

Comment:
Maybe hash the version and the master node ID (that's what an observer considers to be the same for purposes of comparing cluster states)?